### ISSUE TITLE
LoggingMeterRegistry throwing StringIndexOutOfBoundsException in some cases

### ISSUE THREAD

**Issue by killermave** [micrometer-metrics/micrometer#1636]
Hello Folks,
I am using micrometer for getting kafka consumer metrics delivered by LoggingMeterRegistry. I am getting the below exception. I looked at the code and it looks fine to me. Any advise on this is much appreciated.

**Code block in question**
```  String humanReadableByteCount(double bytes) {
            int unit = 1024;
            if (bytes < unit) return decimalOrNan(bytes) + " B";
            int exp = (int) (Math.log(bytes) / Math.log(unit));
            String pre = "KMGTPE".charAt(exp - 1) + "i"; //<-----  exception here
            return decimalOrNan(bytes / Math.pow(unit, exp)) + " " + pre + "B";
        }
```

Some pertinent information

- Kafka Client Version: 2.3.0
- Micrometer Version: 1.3.0
- JDK Version: 1.8.0_191

**Exception Stack trace**
```[logging-metrics-publisher] INFO  io.micrometer.core.instrument.logging.LoggingMeterRegistry - kafka.consumer.fetch.size.avg{client.id=abc,topic=foo} value=3.036592 KiB
  [logging-metrics-publisher] WARN  io.micrometer.core.instrument.push.PushMeterRegistry - Unexpected exception thrown while publishing metrics for LoggingMeterRegistry
 java.lang.StringIndexOutOfBoundsException: String index out of range: -1
 	at java.lang.String.charAt(String.java:658)
 	at io.micrometer.core.instrument.logging.LoggingMeterRegistry$Printer.humanReadableByteCount(LoggingMeterRegistry.java:237)
 	at io.micrometer.core.instrument.logging.LoggingMeterRegistry$Printer.humanReadableBaseUnit(LoggingMeterRegistry.java:244)
 	at io.micrometer.core.instrument.logging.LoggingMeterRegistry$Printer.value(LoggingMeterRegistry.java:229)
 	at io.micrometer.core.instrument.logging.LoggingMeterRegistry.lambda$publish$4(LoggingMeterRegistry.java:121)
 	at io.micrometer.core.instrument.Meter.use(Meter.java:158)
 	at io.micrometer.core.instrument.logging.LoggingMeterRegistry.lambda$publish$13(LoggingMeterRegistry.java:120)
 	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
 	at java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:352)
 	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
 	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
 	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
 	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
 	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
 	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
 	at io.micrometer.core.instrument.logging.LoggingMeterRegistry.publish(LoggingMeterRegistry.java:118)
 	at io.micrometer.core.instrument.push.PushMeterRegistry.publishSafely(PushMeterRegistry.java:48)
 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
 	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
 	at java.lang.Thread.run(Thread.java:748)```

**Comment by checketts** [comment]
Your metric is reporting a NaN. I'll submit a PR to address that.

**Comment by shakuzen** [comment]
Fixed by #1637
